pydrive2/auth.py

```diff
@@ -42,103 +57,6 @@ class RefreshError(AuthError):
     """Access token refresh error."""
 
 
-def LoadAuth(decoratee):
-    """Decorator to check if the auth is valid and loads auth if not."""
-
-    @wraps(decoratee)
-    def _decorated(self, *args, **kwargs):
-        # Initialize auth if needed.
-        if self.auth is None:
-            self.auth = GoogleAuth()
-        # Re-create access token if it expired.
-        if self.auth.access_token_expired:
-            if getattr(self.auth, "auth_method", False) == "service":
-                self.auth.ServiceAuth()
-            else:
-                self.auth.LocalWebserverAuth()
-
-        # Initialise service if not built yet.
-        if self.auth.service is None:
-            self.auth.Authorize()
-
-        # Ensure that a thread-safe HTTP object is provided.
-        if (
```

Comment:
It looks like providing a custom `http` into a request won't work anymore. It might be needed to scenarios like proxy servers, etc. When users take responsibility in providing and managing HTTP clients.

Reply:
Could you share some examples of how this decorator can be used to provide a custom `http` into a request? 

Regarding the previous discussion on caching of Http object per thread, I will add thread_local back to the auth object now that I understand its purpose

Reply:
@junpeng-jp any progress on that?

Reply:
I've added this as part of a new lazily-loaded property called `authorized_http`. It would create a `AuthorizedHttp` object and add it to `self.thread_local` if it doesn't exist. Else, just retrieve it from `self.thread_local`